recipe/session/sessmodels/models.go

```diff
@@ -62,6 +66,51 @@ func (h *HandshakeInfo) SetJwtSigningPublicKeyList(updatedList []KeyInfo) {
 	h.rawJwtSigningPublicKeyList = updatedList
 }
 
+type GetJWKSFunction = func() (*keyfunc.JWKS, error)
+
+func GetJWKS() []GetJWKSFunction {
+	result := []GetJWKSFunction{}
+	corePaths := supertokens.GetAllCoreUrlsForPath("/.well-known/jwks.json")
+
+	for _, path := range corePaths {
+		result = append(result, func() (*keyfunc.JWKS, error) {
+			// RefreshUnknownKID - Fetch JWKS again if the kid in the header of the JWT does not match any in cache
+			// RefreshRateLimit - Only allow one re-fetch every 500 milliseconds
+			// RefreshInterval - Refreshes should occur every 60 seconds
+			jwks, err := keyfunc.Get(path, keyfunc.Options{
+				RefreshUnknownKID: true,
+				RefreshRateLimit:  time.Millisecond * 500,
+				RefreshInterval:   time.Millisecond * time.Duration(JWKCacheMaxAgeInMs),
+			})
+
+			return jwks, err
+		})
+	}
+
+	return result
+}
+
+func GetCombinedJWKS() (*keyfunc.JWKS, error) {
```

Comment:
@porcellus This should be done in Node as well

Reply:
Hmm, I'd explain it a bit differently because all cores have to use the same keys to verify. Otherwise, sessions are not portable between instances, and validation would fail if calling one core but passing in another. I don't think this is an edge case, it doesn't really work even on the core level.